api/src/main/java/org/apache/iceberg/types/CheckCompatibility.java

```diff
@@ -38,7 +38,18 @@
    * @return a list of error details, or an empty list if there are no compatibility problems
    */
   public static List<String> writeCompatibilityErrors(Schema readSchema, Schema writeSchema) {
-    return TypeUtil.visit(readSchema, new CheckCompatibility(writeSchema, true));
+    return TypeUtil.visit(readSchema, new CheckCompatibility(writeSchema, true, true));
+  }
+
+  /**
+   * Returns a list of compatibility errors for writing with the given write schema.
```

Comment:
This doc should be updated to explain how it differs from `writeCompatibilityErrors`. Probably update the doc for that method as well to note that it does check field nullability.

Reply:
👍

Reply:
Similar to above, I think this wording is difficult to understand. How about this instead?

> `This checks type compatibility and not nullability: writing a optional (nullable) values to a required field is not an error. To check nullability as well as types, use {@link #writeCompatibilityErrors(Schema, Schema)}.`

Reply:
@andrei-ionescu, could you update this one as well?

Reply:
👍